accel/tcg/tcg-runtime-sym-vec.c

```diff
@@ -0,0 +1,516 @@
+ *   arg1_symbolic, arg2_symbolic : symbolic expressions of the input vector operands
+ *   comparison_operator : enum value that represents a comparison operator like <, >, ==, etc.
+ *   result_concrete : pointer to a buffer that stores the concrete value of the result vector, as computed by the
+ *       concrete execution of the SIMD operation
+ *   vector_size : size of the vectors in bits
+ *   vece : element size in bits = 8 * 2^vece
+ *
+ */
+void *HELPER(sym_cmp_vec)(
+        CPUArchState *env,
+        void *arg1_concrete, void *arg1_symbolic,
+        void *arg2_concrete, void *arg2_symbolic,
+        TCGCond comparison_operator, void *result_concrete,
+        uint64_t vector_size, uint64_t vece
+) {
+    g_assert(vector_size == 64 || vector_size == 128 || vector_size == 256);
+    uint64_t element_size = vece_element_size(vece);
+    g_assert(element_size <= vector_size);
+    g_assert(vector_size % element_size == 0);
+
+    if (arg1_symbolic == NULL && arg2_symbolic == NULL) {
+        return NULL;
+    }
+
+    if (arg1_symbolic == NULL) {
+        arg1_symbolic = _sym_build_integer_from_buffer(arg1_concrete, vector_size);
+    }
+
+    if (arg2_symbolic == NULL) {
+        arg2_symbolic = _sym_build_integer_from_buffer(arg2_concrete, vector_size);
+    }
+
+    g_assert(_sym_bits_helper(arg1_symbolic) == _sym_bits_helper(arg2_symbolic) &&
+             _sym_bits_helper(arg1_symbolic) == vector_size);
+
+    uint64_t element_count = vector_size / element_size;
+
+    void *arg1_elements[element_count];
+    void *arg2_elements[element_count];
+
+    split_expression(arg1_symbolic, element_count, element_size, arg1_elements);
+    split_expression(arg2_symbolic, element_count, element_size, arg2_elements);
+
+    for (uint64_t i = 0; i < element_count; i++) {
+        /* For each element, the comparison was true iff the element of the result is equal to -1.
+        * Here it is ok to take the first byte of the element, because the result is either -1 or 0. */
+        uint8_t is_taken = *(uint8_t *) element_address(result_concrete, i, element_size, vector_size);
+        build_and_push_path_constraint(
+                env,
+                arg1_elements[i],
+                arg2_elements[i],
+                comparison_operator,
+                is_taken
+        );
+
+    }
+
+    /* Concretize the result, as path constraints for its value have been pushed. */
+    return NULL;
+}
+
+/*
+ * Symbolic equivalent of an SIMD ternary operation.
+ *
+ * This function performs the symbolic equivalent of computing an output vector of the form :
+ * result[i] = arg1[i] <comparison_operator> arg2[i] ? arg1[i] : arg2[i]
+ * where <vector>[i] denotes the ith element of the vector <vector>.
+ *
+ * For each element, the concrete result is used to determine if the ternary condition was true or false,
+ * and path constraints are pushed accordingly.
```

Comment:
Same here, let's add the exploration of a symbolic result instead of path constraints as a TODO comment.